modules/explorer/database.go

```diff
@@ -0,0 +1,105 @@
+package explorer
+
+import (
+	"errors"
+
+	"github.com/NebulousLabs/Sia/encoding"
+	"github.com/NebulousLabs/Sia/types"
+
+	"github.com/NebulousLabs/bolt"
+)
+
+var (
+	errNotExist = errors.New("entry does not exist")
+
+	// database buckets
+	bucketBlockFacts            = []byte("BlockFacts")
+	bucketBlockIDs              = []byte("BlockIDs")
+	bucketBlocksDifficulty      = []byte("BlocksDifficulty")
+	bucketBlockTargets          = []byte("BlockTargets")
+	bucketFileContractHistories = []byte("FileContractHistories")
+	bucketFileContractIDs       = []byte("FileContractIDs")
+	bucketSiacoinOutputIDs      = []byte("SiacoinOutputIDs")
+	bucketSiacoinOutputs        = []byte("SiacoinOutputs")
+	bucketSiafundOutputIDs      = []byte("SiafundOutputIDs")
+	bucketSiafundOutputs        = []byte("SiafundOutputs")
+	bucketTransactionIDs        = []byte("TransactionIDs")
+	bucketUnlockHashes          = []byte("UnlockHashes")
+
+	// bucketInternal is used to store values internal to the explorer
+	bucketInternal = []byte("Internal")
+
+	// keys for bucketInternal
+	internalBlockHeight  = []byte("BlockHeight")
+	internalRecentChange = []byte("RecentChange")
+)
+
+// These functions all return a 'func(*bolt.Tx) error', which, allows them to
+// be called concisely with the db.View and db.Update functions, e.g.:
+//
+//    var height types.BlockHeight
+//    db.View(dbGetAndDecode(bucketBlockIDs, id, &height))
+//
+// Instead of:
+//
+//   var height types.BlockHeight
+//   db.View(func(tx *bolt.Tx) error {
+//       bytes := tx.Bucket(bucketBlockIDs).Get(encoding.Marshal(id))
+//       return encoding.Unmarshal(bytes, &height)
+//   })
+
+// dbGetAndDecode returns a 'func(*bolt.Tx) error' that retrieves and decodes
+// a value from the specified bucket. If the value does not exist,
+// dbGetAndDecode returns errNotExist.
```

Comment:
In the context of the other functions that return a `func(*bolt.Tx) error` this is clear and just the comment more wordy.